oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/index/ElasticIndexHelper.java

```diff
@@ -132,25 +143,22 @@ private static void mapInternalProperties(XContentBuilder mappingBuilder) throws
     }
 
     private static void mapIndexRules(ElasticIndexDefinition indexDefinition, XContentBuilder mappingBuilder) throws IOException {
-        // we need to check if in the defined rules there are properties with the same name and different types
-        final List<Map.Entry<String, List<PropertyDefinition>>> multiTypesFields = indexDefinition.getPropertiesByName()
-                .entrySet()
-                .stream()
-                .filter(e -> e.getValue().size() > 1)
-                .filter(e -> e.getValue().stream().map(PropertyDefinition::getType).distinct().count() > 1)
-                .collect(Collectors.toList());
-
-        if (!multiTypesFields.isEmpty()) {
-            String fields = multiTypesFields.stream().map(Map.Entry::getKey).collect(Collectors.joining(", ", "[", "]"));
-            throw new IllegalStateException(indexDefinition.getIndexPath() + " has properties with the same name and " +
-                    "different types " + fields);
-        }
+        checkIndexRules(indexDefinition);
 
         for (Map.Entry<String, List<PropertyDefinition>> entry : indexDefinition.getPropertiesByName().entrySet()) {
             final String name = entry.getKey();
             final List<PropertyDefinition> propertyDefinitions = entry.getValue();
 
-            Type<?> type = Type.fromTag(propertyDefinitions.get(0).getType(), false);
+            Type<?> type = null;
```

Comment:
is "type" used later? if not, I would move it into the loop

Reply:
yes, `type` is used by the code after the loop